src/pages/workspaces/workspace/workflows/ExportWorkflowModal.tsx

```diff
@@ -0,0 +1,202 @@
+import { Modal, useUniqueId } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import React, { ReactNode, useState } from 'react';
+import { ButtonPrimary, spinnerOverlay } from 'src/components/common';
+import ErrorView from 'src/components/ErrorView';
+import { ValidatedInput } from 'src/components/input';
+import { ExportWorkflowToWorkspaceProvider } from 'src/libs/ajax/workspaces/providers/ExportWorkflowToWorkspaceProvider';
+import { FormLabel } from 'src/libs/forms';
+import * as Utils from 'src/libs/utils';
+import { workflowNameValidation } from 'src/libs/workflow-utils';
+import { useWorkspaces } from 'src/workspaces/common/state/useWorkspaces';
+import { WorkspaceSelector } from 'src/workspaces/common/WorkspaceSelector';
+import { WorkspaceInfo, WorkspaceWrapper } from 'src/workspaces/utils';
+import validate from 'validate.js';
+
+export interface ExportWorkflowModalProps {
+  defaultWorkflowName: string;
+  destinationWorkspace: WorkspaceInfo | ((workspace: WorkspaceWrapper) => boolean);
+  title: string;
+  exportButtonText: string;
+  exportProvider: ExportWorkflowToWorkspaceProvider;
+  onGoToExportedWorkflow?: (selectedWorkspace: WorkspaceInfo, workflowName: string) => void;
+  onSuccess?: () => void;
+  onDismiss: (event: React.MouseEvent | React.KeyboardEvent) => void;
+}
+
+/**
+ * A customizable component for exporting a particular workflow from any source
+ * into an existing workspace.
+ *
+ * The component first displays the "main export modal," which allows the user
+ * to choose a new name for the workflow in its new workspace, and optionally
+ * allows them to select the destination workspace for the export. Once the
+ * export is successfully completed, the component can optionally show a
+ * "post-export modal" that allows the user to choose whether to go to the
+ * location of the new exported workflow.
+ *
+ * @param {ExportWorkflowModalProps} props
```

Comment:
lets add a blank line between @param blocks to help readability

Reply:
Updated!